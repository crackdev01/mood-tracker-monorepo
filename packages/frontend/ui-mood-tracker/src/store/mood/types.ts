// Define Mood Store State.
export interface MoodState {
  mood: [];
}

// Define Mood Store actions.
export enum MoodActions {
  FETCH_MOODS = 'FETCH_MOODS',
  RENDER_MOODS = 'RENDER_MOODS',
}
