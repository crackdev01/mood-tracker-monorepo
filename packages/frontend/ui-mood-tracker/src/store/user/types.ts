export interface UserRequest {
  username: string;
  password: string;
}

export interface Jwt {
  exp: number;
  iat: number;
  username: string;
  uuid: string;
}

export interface Coordinates {
  latitude: number;
  longitude: number;
}

export interface User {
  accessToken: string;
  decodedAccessToken: Jwt | null;
  // We set the coordinate values if user has provided permission.
  // Else we set to false.
  location?: Coordinates | string;
}

// Define User Store State.
export interface UserState {
  user: User;
}

// Define User Store actions.
export enum UserActions {
  FETCH_USER = 'FETCH_USER',
  RENDER_USER = 'RENDER_USER',
  LOGOUT_USER = 'LOGOUT_USER',
  SET_LOCATION_SUCCESS = 'SET_LOCATION_SUCCESS',
  SET_LOCATION_DENIED = 'SET_LOCATION_DENIED',
  SET_LOCATION_ERROR = 'SET_LOCATION_ERROR',
}
