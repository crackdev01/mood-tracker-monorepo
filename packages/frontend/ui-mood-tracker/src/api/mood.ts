import { authMoodTrackerApi } from './core';

// FIXME: update types.
export async function getMoodEntries(): Promise<any> {
  return await authMoodTrackerApi.get('/mood/entries');
}

export async function postMoodEntry(payload: any): Promise<void> {
  return await authMoodTrackerApi.post('/mood/entry', payload);
}

export async function updateMoodEntry(payload: any): Promise<void> {
  return await authMoodTrackerApi.put('/mood/entry', payload);
}

export async function deleteMoodEntry(payload: any): Promise<void> {
  return await authMoodTrackerApi.delete(`/mood/entry/${payload.id}`);
}
