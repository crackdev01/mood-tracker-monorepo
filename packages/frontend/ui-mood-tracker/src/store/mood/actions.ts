import { MoodActions } from './types';

export const fetchMoodEntries = () => {
  return {
    type: MoodActions.FETCH_MOODS,
  };
};

export const addMoodEntry = () => {
  return {
    type: MoodActions.ADD_MOOD,
  };
};

export const editMoodEntry = () => {
  return {
    type: MoodActions.EDIT_MOOD,
  };
};

export const deleteMoodEntry = () => {
  return {
    type: MoodActions.DELETE_MOOD,
  };
};

// FIXME: fix type.
export const renderMoodEntries = (mood: any) => {
  return {
    type: MoodActions.RENDER_MOODS,
    mood,
  };
};
