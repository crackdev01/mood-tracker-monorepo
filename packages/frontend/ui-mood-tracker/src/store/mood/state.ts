import { MoodState } from './types';

export const moodState: MoodState = {
  mood: [],
};
