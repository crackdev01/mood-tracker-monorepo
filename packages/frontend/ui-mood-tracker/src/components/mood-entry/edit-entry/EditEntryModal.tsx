import React, { useEffect, useState } from 'react';
import { Button, Dropdown, Header, Icon, Modal } from 'semantic-ui-react';
import { useTranslation } from 'react-i18next';
import { useDispatch } from 'react-redux';
import { MoodActions } from '../../../store/mood/types';

const EditEntryModal = (props: any) => {
  const dispatch = useDispatch();
  const { t } = useTranslation(['MoodEntry']);
  const [showModal, setShowModal] = useState(false);
  // TODO: Repeated code. Extract it to common file.
  const moodStatusOptions = [
    {
      key: 'relaxed',
      text: t('moodStatus.relaxed'),
      value: 'relaxed',
    },
    {
      key: 'motivated',
      text: t('moodStatus.motivated'),
      value: 'motivated',
    },
    {
      key: 'energetic',
      text: t('moodStatus.energetic'),
      value: 'energetic',
    },
    {
      key: 'curious',
      text: t('moodStatus.curious'),
      value: 'curious',
    },
    {
      key: 'confident',
      text: t('moodStatus.confident'),
      value: 'confident',
    },
  ];
  const intensityOptions = [
    {
      key: '0',
      text: '0',
      value: '0',
    },
    {
      key: '1',
      text: '1',
      value: '1',
    },
    {
      key: '2',
      text: '2',
      value: '2',
    },
    {
      key: '3',
      text: '3',
      value: '3',
    },
    {
      key: '4',
      text: '4',
      value: '4',
    },
  ];
  const [moodStatus, setMoodStatus] = useState('');
  const [moodIntensity, setMoodIntensity] = useState('');

  const addEntry = (_: any, data: any) => {
    setMoodStatus(data.value);
  };

  const addIntensity = (_: any, data: any) => {
    setMoodIntensity(data.value);
  };

  const editMoodEntry = () => {
    const payload = {
      id: props.mood.mood_id,
      status: moodStatus,
      intensity: moodIntensity,
    };
    dispatch({
      type: MoodActions.EDIT_MOOD,
      payload,
    });
    setShowModal(false);
  };

  useEffect(() => {
    const { displayModal, mood } = props;
    setShowModal(displayModal);
    setMoodStatus(mood.mood_status);
    setMoodIntensity(mood.mood_intensity);
  }, [props]);

  return (
    <Modal closeIcon open={showModal} onClose={() => setShowModal(false)}>
      <Header icon="archive" content={t('header')} />
      <Modal.Content>
        <Dropdown
          placeholder={t('moodStatus.placeholder')}
          fluid
          selection
          options={moodStatusOptions}
          onChange={addEntry}
          defaultValue={moodStatus}
        />
        <Dropdown
          placeholder={t('intensity.placeholder')}
          fluid
          selection
          options={intensityOptions}
          onChange={addIntensity}
          defaultValue={moodIntensity}
        />
      </Modal.Content>
      <Modal.Actions>
        <Button color="red" onClick={() => setShowModal(false)}>
          <Icon name="remove" /> No
        </Button>
        <Button color="green" onClick={editMoodEntry}>
          <Icon name="checkmark" /> Yes
        </Button>
      </Modal.Actions>
    </Modal>
  );
};

export default EditEntryModal;
