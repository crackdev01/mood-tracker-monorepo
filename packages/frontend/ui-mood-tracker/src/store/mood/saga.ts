import { all, call, put, takeEvery, fork } from 'redux-saga/effects';

import { MoodActions } from './types';
import { getMoodEntriesApi, postMoodEntry, updateMoodEntry, deleteMoodEntry } from '../../api/mood';

// FIXME: update types in file.

// Fetch mood entries.
export function* getMoodEntriesCall() {
  const { data } = yield call(getMoodEntriesApi);
  yield put({ type: MoodActions.RENDER_MOODS, data });
}

export function* getMoodEntries() {
  yield takeEvery(MoodActions.FETCH_MOODS, getMoodEntriesCall);
}

// Add mood entry.
export function* addMoodEntryCall(action: any) {
  yield call(postMoodEntry, action.payload);
  yield call(getMoodEntriesCall);
}

export function* addMoodEntry() {
  yield takeEvery(MoodActions.ADD_MOOD, addMoodEntryCall);
}

// Update mood entry.
export function* editMoodEntryCall(action: any) {
  yield call(updateMoodEntry, action.payload);
  yield call(getMoodEntries);
}

export function* editMoodEntry() {
  yield takeEvery(MoodActions.EDIT_MOOD, editMoodEntryCall);
}

// Delete mood entry.
export function* removeMoodEntryCall(action: any) {
  yield call(deleteMoodEntry, action.payload);
  yield call(getMoodEntries);
}

export function* removeMoodEntry() {
  yield takeEvery(MoodActions.DELETE_MOOD, removeMoodEntryCall);
}

export function* moodSaga() {
  yield all([fork(getMoodEntries), fork(addMoodEntry), fork(editMoodEntry), fork(removeMoodEntry)]);
}

export default moodSaga;
