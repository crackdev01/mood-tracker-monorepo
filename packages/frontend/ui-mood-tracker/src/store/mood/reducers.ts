import { moodState } from './state';
import { MoodActions } from './types';

export const mood = (state = moodState, action: any) => {
  switch (action.type) {
    case MoodActions.FETCH_MOODS:
      return {
        ...state,
        mood: action.mood,
      };
    default:
      return state;
  }
};
