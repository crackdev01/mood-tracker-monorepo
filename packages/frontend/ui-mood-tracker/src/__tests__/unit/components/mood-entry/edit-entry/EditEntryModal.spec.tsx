import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import EditEntryModal from '../../../../../components/mood-entry/edit-entry/EditEntryModal';
import { mockMoods } from '../../../../fixtures';
import { MoodActions } from '../../../../../store/mood/types';

Enzyme.configure({ adapter: new Adapter() });

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

describe('EditEntryModal', () => {
  const props = { mood: mockMoods[0] };
  test('renders', () => {
    mockDispatch.mockReturnValueOnce({});
    const w = shallow(<EditEntryModal {...props} />);
    expect(w.debug()).toMatchSnapshot();
  });

  test('triggers editMoodEntry', () => {
    mockDispatch.mockReturnValueOnce({});
    const w = shallow(<EditEntryModal {...props} />);
    w.find('Dropdown').at(0).simulate('change', undefined, { value: 'curious' });
    w.find('Dropdown').at(1).simulate('change', undefined, { value: 4 });
    w.find('Button').at(1).simulate('click');
    expect(mockDispatch).toHaveBeenCalledWith({
      type: MoodActions.EDIT_MOOD,
      payload: {
        id: mockMoods[0].mood_id,
        status: 'curious',
        intensity: 4,
      },
    });
  });
});
