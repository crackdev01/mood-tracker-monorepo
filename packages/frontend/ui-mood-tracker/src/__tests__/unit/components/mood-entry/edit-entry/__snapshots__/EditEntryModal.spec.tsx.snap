// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditEntryModal renders 1`] = `
"<Modal closeIcon={true} open={false} onClose={[Function: onClose]} centered={true} dimmer={true} closeOnDimmerClick={true} closeOnDocumentClick={false} eventPool=\\"Modal\\">
  <Header icon=\\"archive\\" content=\\"header\\" />
  <ModalContent>
    <Dropdown placeholder=\\"moodStatus.placeholder\\" fluid={true} selection={true} options={{...}} onChange={[Function: addEntry]} defaultValue=\\"\\" additionLabel=\\"Add \\" additionPosition=\\"top\\" closeOnBlur={true} closeOnEscape={true} deburr={false} icon=\\"dropdown\\" minCharacters={1} noResultsMessage=\\"No results found.\\" openOnFocus={true} renderLabel={[Function: renderItemContent]} searchInput=\\"text\\" selectOnBlur={true} selectOnNavigation={true} wrapSelection={true} />
    <Dropdown placeholder=\\"intensity.placeholder\\" fluid={true} selection={true} options={{...}} onChange={[Function: addIntensity]} defaultValue=\\"\\" additionLabel=\\"Add \\" additionPosition=\\"top\\" closeOnBlur={true} closeOnEscape={true} deburr={false} icon=\\"dropdown\\" minCharacters={1} noResultsMessage=\\"No results found.\\" openOnFocus={true} renderLabel={[Function: renderItemContent]} searchInput=\\"text\\" selectOnBlur={true} selectOnNavigation={true} wrapSelection={true} />
  </ModalContent>
  <ModalActions>
    <Button color=\\"red\\" onClick={[Function: onClick]} as=\\"button\\">
      <Icon name=\\"remove\\" as=\\"i\\" />
       No
    </Button>
    <Button color=\\"green\\" onClick={[Function: editMoodEntry]} as=\\"button\\">
      <Icon name=\\"checkmark\\" as=\\"i\\" />
       Yes
    </Button>
  </ModalActions>
</Modal>"
`;
